Yoast/wordpress-seo
# Look for remote RC branch

## Context


* When creating a new RC. If the RC branch exists remotely, this should be checked out and used. This happens on subsequent RCs which are made by different people.

## Summary


This PR can be summarized in the following changelog entry:

* [not-user-facing] Improvements to the RC automization script

## Relevant technical choices:

* N/A

## Test instructions

This PR can be tested by following these steps:

As this PR switches branches, adjusting the baseBranch is needed to keep the code being present.

Change the `const basebranch = type === "hotfix" ? "master" : "trunk";` line t the following: 
`const basebranch = "14533-look-for-remote-rc-branch";`

To make sure this branch is where you end up when preparing the branches.

* Make sure the release/14.4 does not exist locally (git branch -d release/14.4), which will be checked out from the remote
* grunt ensure-pre-release-branch --plugin-version=14.4 --type=release 

* Make sure a version that is not existing locally or remotely is creating a new branch locally, will be pushed later
* grunt ensure-pre-release-branch --plugin-version=99.99 --type=release                                                                                                          

* Make sure having a local branch, which is not pushed to the remote fails
* git checkout -b release/100-local
* git ensure-pre-release-branch --plugin-version=100-local --type=release

## UI changes

## Documentation

## Quality assurance


Fixes #14533
